invenio_app_rdm/records_ui/searchapp.py

```diff
@@ -7,250 +7,46 @@
-            raise ValueError(
-                'Parameter default_size should be part of pagination_options')
-        return {
-                "resultsPerPage": [
-                    {"text": str(option), "value": option}
-                    for option in self.pagination_options
-                ],
-                "defaultValue": self.default_size,
-            }
-
-    @property
-    def defaultSortingOnEmptyQueryString(self):
-        """Defines the default sorting options when there is no query."""
-        return {
-            "sortBy": self.sort.default_no_query,
-        }
-
-    @classmethod
-    def generate(cls, options, **kwargs):
-        """Create JSON config for React-Searchkit."""
-        generator_object = cls(options)
-        config = {
-            "appId": generator_object.appId,
-            "initialQueryState": generator_object.initialQueryState,
-            "searchApi": generator_object.searchApi,
-            "sortOptions": generator_object.sortOptions,
-            "aggs": generator_object.aggs,
-            "layoutOptions": generator_object.layoutOptions,
-            "sortOrderDisabled": True,
-            "paginationOptions": generator_object.paginationOptions,
-            "defaultSortingOnEmptyQueryString":
-                generator_object.defaultSortingOnEmptyQueryString
-
-        }
-        config.update(kwargs)
-        return config
-
-
-#
-# Application state context processors
-#
-def sort_config(config_name):
-    """Sort configuration."""
-    return SortConfig(
-        current_app.config['RDM_SORT_OPTIONS'],
-        current_app.config[config_name].get('sort', []),
-        current_app.config[config_name].get('sort_default', None),
-        current_app.config[config_name].get('sort_default_no_query'),
-    )
-
-
-def facets_config(config_name, available_facets):
-    """Facets configuration."""
-    return FacetsConfig(
-        available_facets,
-        current_app.config[config_name].get('facets', [])
-    )
-
-
-def search_app_config(
-    config_name, available_facets, endpoint, overrides=None, **kwargs
-):
-    """Search app config."""
-    opts = dict(
-        endpoint=endpoint,
-        headers={"Accept": "application/vnd.inveniordm.v1+json"},
-        grid_view=False,
-        sort=sort_config(config_name),
-        facets=facets_config(config_name, available_facets),
-    )
-    opts.update(kwargs)
-    overrides = overrides or {}
-    return SearchAppConfig.generate(opts, **overrides)
+from invenio_search_ui.searchconfig import search_app_config
 
 
 def search_app_context():
     """Search app context processor."""
     return {
-        'search_app_config': partial(
+        'search_app_rdm_config': partial(
             search_app_config,
             'RDM_SEARCH',
             current_app.config['RDM_FACETS'],
-            '/api/records'),
-        'search_app_user_uploads_config': partial(
+            current_app.config['RDM_SORT_OPTIONS'],
+            '/api/records',
```

Comment:
I am not really a fan of hardcoded URLs, can we get it from any config?

Reply:
I agree, it doesn't look like we can get all these URLs from an existent config (unless I am mistaken), might be worth to create a new one, any suggestion for the name? maybe `RDM_SEARCH_CONFIG_URLS`

Reply:
As discussed, next PR ;)